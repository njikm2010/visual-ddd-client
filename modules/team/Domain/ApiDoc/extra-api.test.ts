import { extraReferenceId, extraApi, parentType } from './extra-api';
import data from './extra-api.data.json';

describe('extraReferenceId', () => {
  it('should extract bracketed strings from a given string', () => {
    const str = 'This is a [1:test] string with [2:multiple] [3:bracketed] words';
    const expected = ['test', 'multiple', 'bracketed'];
    const result = extraReferenceId(str);
    expect(result).toEqual(expected);
  });

  it('should return an empty array if no bracketed strings are found', () => {
    const str = 'This is a test string without any bracketed words';
    const result = extraReferenceId(str);
    expect(result).toEqual([]);
  });
});

test('extraApi', () => {
  expect(extraApi(data as any)).toMatchSnapshot();
});

describe('parentType', () => {
  const identity = (name: string, id: string) => ({ id, name });

  it('should parse reference types correctly', () => {
    const result = parentType('string[ref:123]', identity);
    expect(result).toEqual(['string', { id: '123', name: 'ref' }]);
  });

  it('should throw an error for invalid types', () => {
    expect(() => parentType('string[ref:123', identity)).toThrowError('Invalid type: string[ref:123');
    expect(() => parentType('string[ref:123]]', identity)).toThrowError('Invalid type: string[ref:123]]');
    expect(() => parentType('string[ref:123]extra]', identity)).toThrowError('Invalid type: string[ref:123]extra]');
    expect(() => parentType('string[[ref:123][ref2:456]', identity)).toThrowError(
      'Invalid type: string[[ref:123][ref2:456]'
    );
  });

  it('should handle non-reference types', () => {
    const result = parentType('string', identity);
    expect(result).toEqual(['string']);
  });

  it('should handle multiple reference types', () => {
    const result = parentType('string[ref:123][ref2:456]', identity);
    expect(result).toEqual(['string', { id: '123', name: 'ref' }, { id: '456', name: 'ref2' }]);
  });
});
