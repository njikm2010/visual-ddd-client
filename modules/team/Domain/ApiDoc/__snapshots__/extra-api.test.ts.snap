// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extraApi 1`] = `
{
  "commands": [
    {
      "category": "organizationCreate",
      "description": "",
      "event": {
        "name": "OrganizationCreateEvent",
        "properties": [
          {
            "access": "public",
            "description": "",
            "meta": {},
            "name": "id",
            "title": "组织 ID",
            "type": "Long",
            "uuid": "ed05e7e1-d3c6-4412-b933-c4b0fbebe9af",
          },
        ],
        "uuid": "a8f2d25d-00cd-421d-869e-a85f8d882d39",
      },
      "eventSendable": false,
      "meta": {},
      "name": "OrganizationCreate",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "name",
          "title": "名称",
          "type": "String",
          "uuid": "04db7cde-1692-44a6-a256-117c6c573415",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "description",
          "title": "描述",
          "type": "String",
          "uuid": "dc7d5665-c222-4206-b90f-281d5e362cac",
        },
        {
          "access": "public",
          "description": "组织的管理者，被系统管理员授权，管理自己所在组织下的所有团队。",
          "meta": {},
          "name": "organizationManagerId",
          "title": "组织管理员",
          "type": "Long",
          "uuid": "78b4e1cd-0fea-46b6-b9df-4f8668d5caa8",
        },
      ],
      "repository": "create",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [
        {
          "description": "校验名称是否重复",
          "meta": {},
          "name": "CheckRepeatedName",
          "title": "校验名称是否重复",
          "uuid": "1aa8782e-5828-446b-ace2-8c7ca5905ac1",
        },
      ],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "新增组织",
      "type": "Command",
      "uuid": "a8f2d25d-00cd-421d-869e-a85f8d882d39",
    },
    {
      "category": "organizationModify",
      "description": "",
      "event": {
        "name": "OrganizationModifyEvent",
        "properties": [],
        "uuid": "974b9b21-88a3-4b64-bb13-d7017ba4bb8d",
      },
      "eventSendable": false,
      "meta": {},
      "name": "OrganizationModify",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "组织 ID",
          "type": "Long",
          "uuid": "0176edd8-c8d1-485a-ac1e-e4c079e97eba",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "name",
          "title": "名称",
          "type": "String",
          "uuid": "16002449-ed15-4efe-a1ad-9f652f0d6f4b",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "description",
          "title": "描述",
          "type": "String",
          "uuid": "e25fca53-8a04-4d5e-92ff-4fac54263334",
        },
        {
          "access": "public",
          "description": "组织的管理者，被系统管理员授权，管理自己所在组织下的所有团队。",
          "meta": {},
          "name": "organizationManagerId",
          "title": "组织管理员",
          "type": "Long",
          "uuid": "51faa759-c339-48f9-874c-067ebf913828",
        },
      ],
      "repository": "modify",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [
        {
          "description": "校验名称是否重复",
          "meta": {},
          "name": "CheckRepeatedName",
          "title": "校验名称是否重复",
          "uuid": "copy-fa26ba23-1a8d-4b74-873e-3bbef56635a3",
        },
      ],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "编辑组织",
      "type": "Command",
      "uuid": "974b9b21-88a3-4b64-bb13-d7017ba4bb8d",
    },
    {
      "category": "organizationRemove",
      "description": "",
      "event": {
        "name": "OrganizationRemoveEvent",
        "properties": [],
        "uuid": "31a56395-a105-45fe-b48b-d07d5be71ea4",
      },
      "eventSendable": false,
      "meta": {},
      "name": "OrganizationRemove",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "组织 ID",
          "type": "Long",
          "uuid": "38061f3c-0c66-4c39-bf52-cc785accc8d4",
        },
      ],
      "repository": "remove",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [
        {
          "description": "检查组织下是否存在团队",
          "meta": {},
          "name": "CheckExistTeam",
          "title": "存在团队不能删除组织",
          "uuid": "f5867523-0f28-4144-addf-a4f06403c84d",
        },
      ],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "删除组织",
      "type": "Command",
      "uuid": "31a56395-a105-45fe-b48b-d07d5be71ea4",
    },
    {
      "category": "organizationManagerBind",
      "description": "",
      "event": {
        "name": "OrganizationManagerBindEvent",
        "properties": [],
        "uuid": "967c4491-6176-4633-856f-a74c29219e84",
      },
      "eventSendable": false,
      "meta": {},
      "name": "OrganizationManagerBind",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "组织 ID",
          "type": "Long",
          "uuid": "6eef17e3-ab7d-493f-b8d9-105b68d5f47d",
        },
        {
          "access": "public",
          "description": "组织的管理者，被系统管理员授权，管理自己所在组织下的所有团队。",
          "meta": {},
          "name": "organizationManagerId",
          "title": "组织管理员",
          "type": "Long",
          "uuid": "584107c5-7186-4b1a-8921-114e47102654",
        },
      ],
      "repository": "modify",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "绑定组织管理员",
      "type": "Command",
      "uuid": "967c4491-6176-4633-856f-a74c29219e84",
    },
    {
      "category": "organizationManagerUnbind",
      "description": "",
      "event": {
        "name": "OrganizationManagerUnbindEvent",
        "properties": [],
        "uuid": "copy-6ff57305-9236-47a4-aa35-d701c724ce7a",
      },
      "eventSendable": false,
      "meta": {},
      "name": "OrganizationManagerUnbind",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "组织 ID",
          "type": "Long",
          "uuid": "6eef17e3-ab7d-493f-b8d9-105b68d5f47d",
        },
      ],
      "repository": "modify",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "解绑组织管理员",
      "type": "Command",
      "uuid": "copy-6ff57305-9236-47a4-aa35-d701c724ce7a",
    },
    {
      "category": "teamCreate",
      "description": "",
      "event": {
        "name": "TeamCreateEvent",
        "properties": [
          {
            "access": "public",
            "description": "",
            "meta": {},
            "name": "id",
            "title": "团队 ID",
            "type": "Long",
            "uuid": "4f1cea6d-0fe8-4cab-8189-5d449b7b9274",
          },
        ],
        "uuid": "copy-711a540d-2af8-4a8f-9bf4-61bd76859909",
      },
      "eventSendable": false,
      "meta": {},
      "name": "TeamCreate",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "name",
          "title": "名称",
          "type": "String",
          "uuid": "527a9fe0-94a2-48ad-a7df-c23cd5ba67af",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "description",
          "title": "描述",
          "type": "String",
          "uuid": "ff90e119-0293-4bc5-8a18-1ce2e5141513",
        },
        {
          "access": "public",
          "description": "组织唯一标识",
          "meta": {},
          "name": "organizationId",
          "title": "组织ID",
          "type": "Long",
          "uuid": "9f11d0bc-a907-40ad-bd9e-e712992d58ab",
        },
        {
          "access": "public",
          "description": "团队的管理者，被组织管理员授权，管理自己所在团队",
          "meta": {},
          "name": "teamManagerId",
          "title": "团队管理员",
          "type": "Long",
          "uuid": "d7280bbe-567a-4f14-bb01-a6070bbccf12",
        },
      ],
      "repository": "create",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [
        {
          "description": "校验名称是否重复",
          "meta": {},
          "name": "CheckRepeatedName",
          "title": "校验名称是否重复",
          "uuid": "e3178307-1429-48aa-b72e-7d4199847804",
        },
      ],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "新增团队",
      "type": "Command",
      "uuid": "copy-711a540d-2af8-4a8f-9bf4-61bd76859909",
    },
    {
      "category": "teamModify",
      "description": "",
      "event": {
        "name": "TeamModifyEvent",
        "properties": [],
        "uuid": "copy-73391b49-6571-4fc1-823a-04d1bd7ef26a",
      },
      "eventSendable": false,
      "meta": {},
      "name": "TeamModify",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "团队 ID",
          "type": "Long",
          "uuid": "e4aac776-2bf2-42b0-8407-5a5f3319cdca",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "name",
          "title": "名称",
          "type": "String",
          "uuid": "8468dd10-4215-4d39-91c2-8caf1d4d6e9b",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "description",
          "title": "描述",
          "type": "String",
          "uuid": "8828e506-1f49-4bd4-96bc-6e4ba58bbe89",
        },
        {
          "access": "public",
          "description": "组织唯一标识",
          "meta": {},
          "name": "organizationId",
          "title": "组织ID",
          "type": "Long",
          "uuid": "318f0dea-2abd-4b00-94f7-3e1f00f478b2",
        },
        {
          "access": "public",
          "description": "团队的管理者，被组织管理员授权，管理自己所在团队",
          "meta": {},
          "name": "teamManagerId",
          "title": "团队管理员",
          "type": "Long",
          "uuid": "7f118c74-8e86-4a93-965f-b050f0fae0fe",
        },
      ],
      "repository": "modify",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [
        {
          "description": "校验名称是否重复",
          "meta": {},
          "name": "CheckRepeatedName",
          "title": "校验名称是否重复",
          "uuid": "copy-e9c53073-2c11-49b3-8c1c-1ae5fc2ae4a4",
        },
      ],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "编辑团队",
      "type": "Command",
      "uuid": "copy-73391b49-6571-4fc1-823a-04d1bd7ef26a",
    },
    {
      "category": "teamRemove",
      "description": "",
      "event": {
        "name": "TeamRemoveEvent",
        "properties": [],
        "uuid": "copy-f658dc1a-46e7-4e60-b488-ecf78c7242ed",
      },
      "eventSendable": false,
      "meta": {},
      "name": "TeamRemove",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "团队 ID",
          "type": "Long",
          "uuid": "2ec6074e-74f5-4efa-bf7f-700ada96c07b",
        },
      ],
      "repository": "remove",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [
        {
          "description": "检查团队下是否存在应用",
          "meta": {},
          "name": "CheckExistApplication",
          "title": "存在应用不能删除",
          "uuid": "6222037b-dcf2-4c7f-9a91-eb1bd06be896",
        },
        {
          "description": "检查团队下是否存在业务场景",
          "meta": {},
          "name": "CheckExistBusinessScene",
          "title": "存在业务场景不能删除",
          "uuid": "copy-2c557447-f223-4a7d-8674-bf77a4f02159",
        },
        {
          "description": "检查团队下是否存在业务域",
          "meta": {},
          "name": "CheckExistDomainDesign",
          "title": "存在业务域不能删除",
          "uuid": "copy-5657fd45-523b-40ec-9aab-cccae0044e86",
        },
      ],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "删除团队",
      "type": "Command",
      "uuid": "copy-f658dc1a-46e7-4e60-b488-ecf78c7242ed",
    },
    {
      "category": "teamManagerBind",
      "description": "",
      "event": {
        "name": "TeamManagerBindEvent",
        "properties": [],
        "uuid": "c19b7203-3de6-4b7b-a7eb-fc647e2e8937",
      },
      "eventSendable": false,
      "meta": {},
      "name": "TeamManagerBind",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "团队 ID",
          "type": "Long",
          "uuid": "2ef9dd00-04b2-4dda-83e5-fb829df44886",
        },
        {
          "access": "public",
          "description": "团队的管理者，被组织管理员授权，管理自己所在团队",
          "meta": {},
          "name": "teamManagerId",
          "title": "团队管理员",
          "type": "Long",
          "uuid": "3a3ce794-e409-4826-a14d-f1ce5643d0c8",
        },
      ],
      "repository": "modify",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "绑定团队管理员",
      "type": "Command",
      "uuid": "c19b7203-3de6-4b7b-a7eb-fc647e2e8937",
    },
    {
      "category": "teamManagerUnbind",
      "description": "",
      "event": {
        "name": "TeamManagerUnBindEvent",
        "properties": [],
        "uuid": "76210e50-42c2-44a0-b6bf-6e236c5d06b7",
      },
      "eventSendable": false,
      "meta": {},
      "name": "TeamManagerUnBind",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "主键",
          "type": "Long",
          "uuid": "1285be00-99ad-4931-9f27-1348aca6c5e0",
        },
      ],
      "repository": "modify",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "解绑团队管理员",
      "type": "Command",
      "uuid": "76210e50-42c2-44a0-b6bf-6e236c5d06b7",
    },
    {
      "category": "teamMemberAdd",
      "description": "",
      "event": {
        "name": "TeamMemberAddEvent",
        "properties": [],
        "uuid": "7faf7f22-3e8a-4b11-977b-6b76a6b5cb99",
      },
      "eventSendable": false,
      "meta": {},
      "name": "TeamMemberAdd",
      "properties": [
        {
          "access": "public",
          "description": "团队成员的账号唯一标识",
          "meta": {},
          "name": "accountId",
          "title": "成员账号ID",
          "type": "Long",
          "uuid": "ae7468cb-8ea2-440f-b5ff-0bf476047965",
        },
        {
          "access": "public",
          "description": "团队唯一标识",
          "meta": {},
          "name": "teamId",
          "title": "团队ID",
          "type": "Long",
          "uuid": "f151cc0a-2eb0-48af-a9e8-7fd14d0746a3",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "memberTypeList",
          "title": "成员类型标签",
          "type": "List<[MemberType:4eb5926e-77bb-4665-badd-a3bb0d8b763e]>",
          "uuid": "662a46ed-3f7f-4a8a-8d48-591201db11c8",
        },
      ],
      "repository": "create",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [
        {
          "description": "校验团队成员是否重复",
          "meta": {},
          "name": "CheckRepeatedTeamMember",
          "title": "团队成员不能重复",
          "uuid": "49594627-7d0e-45fb-a5e1-db92f3aae9dd",
        },
      ],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "添加团队成员",
      "type": "Command",
      "uuid": "7faf7f22-3e8a-4b11-977b-6b76a6b5cb99",
    },
    {
      "category": "teamMemberRemove",
      "description": "",
      "event": {
        "name": "TeamMemberRemoveEvent",
        "properties": [],
        "uuid": "a559db3f-3012-43f9-97bf-16bde3732b11",
      },
      "eventSendable": false,
      "meta": {},
      "name": "TeamMemberRemove",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "团队成员ID",
          "type": "Long",
          "uuid": "a99c7a69-bdeb-464a-afd7-f6324b762e5c",
        },
      ],
      "repository": "remove",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "删除团队成员",
      "type": "Command",
      "uuid": "a559db3f-3012-43f9-97bf-16bde3732b11",
    },
    {
      "category": "teamMemberRoleBind",
      "description": "",
      "event": {
        "name": "TeamMemberRoleBindEvent",
        "properties": [],
        "uuid": "f07914d8-7023-4b50-bbad-93c6e4e97dbc",
      },
      "eventSendable": false,
      "meta": {},
      "name": "TeamMemberRoleBind",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "团队成员ID",
          "type": "Long",
          "uuid": "1d931112-9bcc-4b4b-8888-a10a7af963c9",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "memberTypeList",
          "title": "成员类型标签",
          "type": "List<[MemberType:4eb5926e-77bb-4665-badd-a3bb0d8b763e]>",
          "uuid": "c7a43335-3e9c-4b94-b0cf-a3fe35ce7b89",
        },
      ],
      "repository": "modify",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "绑定团队成员职位",
      "type": "Command",
      "uuid": "f07914d8-7023-4b50-bbad-93c6e4e97dbc",
    },
    {
      "category": "teamManagerRoleUnbind",
      "description": "",
      "event": {
        "name": "TeamMemberRoleUnBindEvent",
        "properties": [],
        "uuid": "copy-f7fccdac-d38f-4ba0-9afc-749d85297b8c",
      },
      "eventSendable": false,
      "meta": {},
      "name": "TeamMemberRoleUnBind",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "团队成员ID",
          "type": "Long",
          "uuid": "1d931112-9bcc-4b4b-8888-a10a7af963c9",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "memberTypeList",
          "title": "成员类型标签",
          "type": "List<[MemberType:4eb5926e-77bb-4665-badd-a3bb0d8b763e]>",
          "uuid": "c7a43335-3e9c-4b94-b0cf-a3fe35ce7b89",
        },
      ],
      "repository": "modify",
      "result": "void",
      "return": {
        "type": "void",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "解绑团队成员职位",
      "type": "Command",
      "uuid": "copy-f7fccdac-d38f-4ba0-9afc-749d85297b8c",
    },
  ],
  "queries": [
    {
      "description": "查询组织详情",
      "meta": {},
      "name": "OrganizationQuery",
      "pagination": false,
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "组织 ID",
          "type": "Long",
          "uuid": "c2866a4f-0576-49de-88e2-a6a477061e21",
        },
      ],
      "result": "[OrganizationDTO:06a2c1e8-8d27-4c0a-bc26-1f95602c5a48]",
      "return": {
        "type": "[OrganizationDTO:06a2c1e8-8d27-4c0a-bc26-1f95602c5a48]",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "组织详情查询对象",
      "type": "Query",
      "uuid": "595e70ba-0c85-4405-a8a2-04736be4119e",
    },
    {
      "description": "分页查询组织",
      "meta": {},
      "name": "OrganizationPageQuery",
      "pagination": true,
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "name",
          "title": "组织名称",
          "type": "String",
          "uuid": "ad759e5f-4da8-4d16-a413-df4cd823b9e2",
        },
      ],
      "result": "[OrganizationDTO:06a2c1e8-8d27-4c0a-bc26-1f95602c5a48]",
      "return": {
        "type": "[OrganizationDTO:06a2c1e8-8d27-4c0a-bc26-1f95602c5a48]",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "组织分页查询对象 (分页)",
      "type": "Query",
      "uuid": "f18f6eed-9262-41e8-8f8a-192d7b7c51cd",
    },
    {
      "description": "查询团队详情",
      "meta": {},
      "name": "TeamQuery",
      "pagination": false,
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "团队ID",
          "type": "Long",
          "uuid": "e8d30abe-8694-45f1-8c14-b12c3bd359e0",
        },
      ],
      "result": "[TeamDTO:9bbc1892-0bf7-4460-b6c2-f47d40ad210e]",
      "return": {
        "type": "[TeamDTO:9bbc1892-0bf7-4460-b6c2-f47d40ad210e]",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "团队详情查询对象",
      "type": "Query",
      "uuid": "a18d70b1-c07f-4551-b2ef-883ebbdfbf78",
    },
    {
      "description": "分页查询团队",
      "meta": {},
      "name": "TeamPageQuery",
      "pagination": true,
      "properties": [
        {
          "access": "public",
          "description": "组织唯一标识",
          "meta": {},
          "name": "organizationId",
          "title": "组织ID",
          "type": "Long",
          "uuid": "88a094c0-ef87-456b-bab3-930451c45d02",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "name",
          "title": "团队名称",
          "type": "String",
          "uuid": "a2e347b6-a81c-4a3e-93c1-d1a45d4e2baa",
        },
      ],
      "result": "[TeamDTO:9bbc1892-0bf7-4460-b6c2-f47d40ad210e]",
      "return": {
        "type": "[TeamDTO:9bbc1892-0bf7-4460-b6c2-f47d40ad210e]",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "团队分页查询对象 (分页)",
      "type": "Query",
      "uuid": "93b35927-7e63-4fb2-9d92-5173e897d9fc",
    },
    {
      "description": "分页查询团队成员",
      "meta": {},
      "name": "TeamMemberByTeamIdPageQuery",
      "pagination": true,
      "properties": [
        {
          "access": "public",
          "description": "团队唯一标识",
          "meta": {},
          "name": "teamId",
          "title": "团队ID",
          "type": "Long",
          "uuid": "74491d69-3ab5-46d4-854b-1547842cd075",
        },
      ],
      "result": "[TeamMemberDTO:0d948c5a-d87b-458e-9566-a20058bcabe9]",
      "return": {
        "type": "[TeamMemberDTO:0d948c5a-d87b-458e-9566-a20058bcabe9]",
      },
      "rules": [],
      "source": [
        {
          "type": "http",
        },
        {
          "type": "rpc",
        },
      ],
      "title": "查询团队下的成员信息 (分页)",
      "type": "Query",
      "uuid": "c7e8372f-85dc-493e-89e8-e14bd250e71f",
    },
  ],
  "references": {
    "06a2c1e8-8d27-4c0a-bc26-1f95602c5a48": {
      "description": "用于组织的数据传输对象",
      "meta": {},
      "name": "OrganizationDTO",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "组织 ID",
          "type": "Long",
          "uuid": "a7237916-6acd-4d5b-84ed-c493f2ec943a",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "name",
          "title": "名称",
          "type": "String",
          "uuid": "5edf83ff-b1f2-4ce7-850d-3e1f44d51ed9",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "description",
          "title": "描述",
          "type": "String",
          "uuid": "a7657045-22e2-4574-bd1d-cac85dc13923",
        },
        {
          "access": "public",
          "description": "组织的管理者，被系统管理员授权，管理自己所在组织下的所有团队。",
          "meta": {},
          "name": "organizationManagerId",
          "title": "组织管理员",
          "type": "Long",
          "uuid": "cfc1cd0f-774f-4b1b-a5a7-76ae123882a7",
        },
      ],
      "title": "组织数据传输对象",
      "type": "Object",
      "uuid": "06a2c1e8-8d27-4c0a-bc26-1f95602c5a48",
    },
    "0d948c5a-d87b-458e-9566-a20058bcabe9": {
      "description": "用于团队成员的数据传输对象",
      "meta": {},
      "name": "TeamMemberDTO",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "团队成员ID",
          "type": "Long",
          "uuid": "39ef299a-cc63-4508-9d97-7c98a6b5e475",
        },
        {
          "access": "public",
          "description": "团队唯一标识",
          "meta": {},
          "name": "teamId",
          "title": "团队ID",
          "type": "Long",
          "uuid": "9cdac279-09b5-4d47-8e29-5b979bc9de9b",
        },
        {
          "access": "public",
          "description": "团队成员的账号唯一标识",
          "meta": {},
          "name": "accountId",
          "title": "成员账号ID",
          "type": "Long",
          "uuid": "4c27adb9-eb56-4b89-8f55-84eb6a47b09b",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "memberTypeList",
          "title": "成员类型标签",
          "type": "List<Integer>",
          "uuid": "e72462b1-2f56-4c8b-aa3d-79122e1b8d8a",
        },
      ],
      "title": "团队成员数据传输对象",
      "type": "Object",
      "uuid": "0d948c5a-d87b-458e-9566-a20058bcabe9",
    },
    "4eb5926e-77bb-4665-badd-a3bb0d8b763e": {
      "baseType": "number",
      "description": "团队成员在所属团队中扮演的职位",
      "members": [
        {
          "code": 1,
          "description": "需求发起者和验收者，主要负责业务域和业务场景需求的输入、验收项目接口和参与领域建模",
          "meta": {},
          "name": "PM",
          "title": "产品经理",
          "uuid": "75e1a790-5d96-47b4-88eb-48d777e55eaf",
        },
        {
          "code": 0,
          "description": "应用的管理者和领域建模的监督者，主要负责应用的管理和发布、审核领域建模和参与领域建模",
          "meta": {},
          "name": "ARCHITECT",
          "title": "架构师",
          "uuid": "96d3e1c0-6442-406f-9743-d29f274d6b00",
        },
        {
          "code": 0,
          "description": "包括后端开发、前端开发、测试等具体的项目开发人员；主要负责功能开发和参与领域建模",
          "meta": {},
          "name": "DEVELOPER",
          "title": "开发者",
          "uuid": "d488c89f-2227-4331-ab2a-4f7b5c4175eb",
        },
      ],
      "meta": {},
      "name": "MemberType",
      "properties": [
        {
          "code": 1,
          "description": "需求发起者和验收者，主要负责业务域和业务场景需求的输入、验收项目接口和参与领域建模",
          "meta": {},
          "name": "PM",
          "title": "1: 产品经理",
          "type": "Integer",
          "uuid": "75e1a790-5d96-47b4-88eb-48d777e55eaf",
        },
        {
          "code": 0,
          "description": "应用的管理者和领域建模的监督者，主要负责应用的管理和发布、审核领域建模和参与领域建模",
          "meta": {},
          "name": "ARCHITECT",
          "title": "0: 架构师",
          "type": "Integer",
          "uuid": "96d3e1c0-6442-406f-9743-d29f274d6b00",
        },
        {
          "code": 0,
          "description": "包括后端开发、前端开发、测试等具体的项目开发人员；主要负责功能开发和参与领域建模",
          "meta": {},
          "name": "DEVELOPER",
          "title": "0: 开发者",
          "type": "Integer",
          "uuid": "d488c89f-2227-4331-ab2a-4f7b5c4175eb",
        },
      ],
      "title": "团队成员类型",
      "type": "Enum",
      "uuid": "4eb5926e-77bb-4665-badd-a3bb0d8b763e",
    },
    "9bbc1892-0bf7-4460-b6c2-f47d40ad210e": {
      "description": "用于团队的数据传输对象",
      "meta": {},
      "name": "TeamDTO",
      "properties": [
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "id",
          "title": "团队ID",
          "type": "Long",
          "uuid": "371e4750-2032-44a5-b995-a2df477a3951",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "name",
          "title": "名称",
          "type": "String",
          "uuid": "9e2dd2de-185c-4b57-a5dc-d82aedc64982",
        },
        {
          "access": "public",
          "description": "",
          "meta": {},
          "name": "description",
          "title": "描述",
          "type": "String",
          "uuid": "163e1547-26ac-4efb-9b63-3735d5b8dd3d",
        },
        {
          "access": "public",
          "description": "组织唯一标识",
          "meta": {},
          "name": "organizationId",
          "title": "组织ID",
          "type": "Long",
          "uuid": "f2a32b75-8b10-4bb0-8b29-055117b91542",
        },
        {
          "access": "public",
          "description": "团队的管理者，被组织管理员授权，管理自己所在团队",
          "meta": {},
          "name": "teamManagerId",
          "title": "团队管理员",
          "type": "Long",
          "uuid": "82eeb218-2f4b-4aaf-a38d-7ced833405bf",
        },
        {
          "access": "public",
          "description": "团队成员",
          "meta": {},
          "name": "teamMembers",
          "title": "团队成员",
          "type": "List<[TeamMemberDTO:0d948c5a-d87b-458e-9566-a20058bcabe9]>",
          "uuid": "c43a9a49-1207-462a-8f0f-e740d67f1e86",
        },
      ],
      "title": "团队数据传输对象",
      "type": "Object",
      "uuid": "9bbc1892-0bf7-4460-b6c2-f47d40ad210e",
    },
  },
}
`;
